cmake/compiler_support.cmake

```diff
@@ -0,0 +1,261 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2017, 2017
+#
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License v1.0 and
+#  Apache License v2.0 which accompanies this distribution.
+#
+#      The Eclipse Public License is available at
+#      http://www.eclipse.org/legal/epl-v10.html
+#
+#      The Apache License v2.0 is available at
+#      http://www.opensource.org/licenses/apache2.0.php
+#
+# Contributors:
+#    Multiple authors (IBM Corp.) - initial implementation and documentation
+###############################################################################
+
+# This file contains a number of support pieces required to build the compiler
+# component. 
+# 
+# One of the building blocks of this support is the notion that each compiler
+# you may want to build has an associated 'name'. For example jitbuilder is 
+# the name of one. 
+
+## Tools and helpers:
+
+Find_package(Perl)
+
+if (NOT PERL_FOUND )
+	message(FATAL_ERROR "Perl not found")
+endif()
+
+set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
+
+# Given a prefix, and a list of arguments, prefix the list of arguments and
+# assign to out: ie, addprefix(out "-I" "a;b;c") should set out to
+# "-Ia;-Ib;-Ic".
+function(addprefix out prefix) 
+   set(ret "")
+   foreach(var IN ITEMS ${ARGN}) 
+      list(APPEND ret "${prefix}${var}")
+   endforeach()
+   set(${out} ${ret} PARENT_SCOPE)
+endfunction(addprefix) 
+
+
+# Platform setup code!
+# TODO: THis needs to be abstracted for cross platform builds
+
+set(TR_TARGET_ARCH    x     CACHE INTERNAL "The architecture directory used for the compiler code (x, p, or z)")
+set(TR_TARGET_SUBARCH amd64 CACHE INTERNAL "The subarchitecture directory used for the compiler code. May be empty (i386 or amd64)")
+set(CMAKE_ASM-ATT_FLAGS "--64 --defsym TR_HOST_X86=1 --defsym TR_HOST_64BIT=1 --defsym BITVECTOR_64BIT=1 --defsym LINUX=1 --defsym TR_TARGET_X86=1 --defsym TR_TARGET_64BIT=1" CACHE INTERNAL "ASM FLags")
+
+# Mark a target as consuming the compiler components. 
+# 
+# This supplements the compile definitions, options and 
+# include directories.
+# 
+# Call as make_compiler_component(<target> COMPILER <compiler name>)
+function(make_compiler_target TARGET_NAME) 
+   cmake_parse_arguments(TARGET 
```

Comment:
Partially this is reflecting the development history. At one point, I believe I planned for this to be more complex. However, that complexity turned out not to be needed. 

Having said that, I prefer named argument style for CMake myself, as generally it makes reading easier IMO.